cache-filter/src/filter/http.rs

```diff
@@ -115,36 +118,33 @@ impl CacheFilter {
         true
     }
 
-    fn is_rate_limited(&mut self, app: &RefCell<Application>, _current_time: &Duration) -> bool {
+    fn is_rate_limited(&mut self, app: &RefCell<Application>, current_time: &Duration) -> bool {
         for (metric, hits) in self.req_data.metrics.borrow().iter() {
-            // Check metric is present inside local cache
-            if !app.borrow().local_state.borrow().contains_key(metric) {
-                continue;
-            }
-
-            /* Check period window expiration
-            if app.local_state.borrow().get(metric).unwrap().period_window.end < current_time {
-                // Period window is expired
-                // Update period window using get_new_window for each metric
-                // But first confirm how to deal with time sync between host and 3scale
-                // Reset left_hits to max value
-            }*/
+            if let Some(usage_report) = app.borrow_mut().local_state.get_mut(metric) {
+                let mut period = &mut usage_report.period_window;
+                if period.window != Period::Eternity && period.end < *current_time {
+                    // Taking care of period window expiration
+                    let num_windows = current_time.checked_sub(period.start).unwrap().as_secs()
```

Comment:
What you might want to do regarding these functions that return potentially an `Option::None` or `Result::Err` is use a library like `thiserror` to define a custom error type per module that can be built using other types of errors - and can also include error descriptions of your own -, and then change the signature of a function to return ie. `Result<bool, RateLimitedError>` and use the `?` operator to cleanly get rid of the unhappy paths. See https://github.com/3scale/threescale-wasm-auth/blob/d207f09084ebed548de49adf2c937a32d0302bf3/src/configuration/operation/decode.rs for an example (and notice the `?` usage in the code to automatically return the Error type declared on top by building it out of whatever the expression the `?` applies to).

Reply:
As I was writing the patch for `threescale-wasm-auth`, I noticed how simple this pattern is and used it extensively inside the PR. Also, changed the code to use this pattern.